Jenkinsfile

```diff
@@ -51,6 +51,10 @@ spec:
 
 /** Returns the download URL of the JDK against whoose C headers (in the 'include/' folder) and native libaries the SWT natives are compiled.*/
 def getNativeJdkUrl(String os, String arch){ // To update the used JDK version update the URL template below
+	//TODO: remove this workaround as soon as Temurin provides aarch64 JDKs for Windows
+	if('win32'.equals(os) && 'aarch64'.equals(arch)){
+		arch = 'x86_64'
+	}
```

Comment:
@HannesWell I don't think this "aarch64 replacement" with the x86_64 equivalent will work; it may work for the compiling step, but will certainly fail during linking due to mismatched native libs. If it does work during linking, which I very much doubt it, then the resulting `.dll` files won't work at runtime on a Windows Arm64 box.

Reply:
@chirontt How did you get it working on your side? Did you use Microsoft's OpenJDK?

Reply:
> @HannesWell I don't think this "aarch64 replacement" with the x86_64 equivalent will work; it may work for the compiling step, but will certainly fail during linking due to mismatched native libs. If it does work during linking, which I very much doubt it, then the resulting `.dll` files won't work at runtime on a Windows Arm64 box.

Yes. As we can see in the logs of the latest build you anticipated it right and it does not even build.
But this was just temporary to reach the the part of the pipeline where the new native agent is contacted and to test its very basic functionality (quasi a hello-world). But all I see looks very promising. We even reached to point that the native build complains that the JDK-libs don't match the architecture (the anticipated error):

```
   Creating library swt-awt-win32-4965r4.lib and object swt-awt-win32-4965r4.exp
swt_awt.obj : error LNK2019: unresolved external symbol __imp_JAWT_GetAWT referenced in function Java_org_eclipse_swt_awt_SWT_1AWT_getAWTHandle
C:\Users\genie.releng.rie8t-win11-arm\jenkins_agent\workspace\eclipse.platform.swt_PR-1045\jdk.resources\lib\jawt.lib : warning LNK4272: library machine type 'x64' conflicts with target machine type 'ARM64'
LINK : error LNK1218: warning treated as error; no output file generated
NMAKE : fatal error U1077: 'link @templrf' : return code '0x10b0'
Stop.
```
I'll continue tomorrow and adjust the pipeline to use the JDK installed at the agent (which I assume is a WoA one) for compilation.

Eventually Adoptium should provide also JDKs for WoA (not only for SWT but for WoA support of Eclipse in general)
- https://github.com/eclipse-justj/justj/issues/26
- https://github.com/adoptium/adoptium-support/issues/616

Reply:
@hmartinez82 Yes, I had used the Microsoft's OpenJDK build for Arm64 for over a year without any issues, but lately I've started using the beta version of OpenJDK Temurin build for Arm64 here: https://github.com/adoptium/temurin17-binaries/releases/tag/jdk17u-2024-02-07-14-14-beta

@HannesWell You can use the Microsoft OpenJDK build for Arm64 [here](https://learn.microsoft.com/en-us/java/openjdk/download) as a temporary solution for Arm64, while waiting for the official OpenJDK Temurin Arm64 release. But if you insist on using a Temurin OpenJDK build, you can try a beta version of it for Arm64 mentioned above.

Reply:
> lately I've started using the beta version of OpenJDK Temurin build for Arm64 here: https://github.com/adoptium/temurin17-binaries/releases/tag/jdk17u-2024-02-07-14-14-beta

Thanks for that hint. I used that for the moment in order to verify it works and it does.
We had the first successful build of the SWT native binaries for Windows on ARM. 🎉